libs/x-content/impl/src/test/java/org/elasticsearch/xcontent/provider/json/ESUTF8StreamJsonParserTests.java

```diff
@@ -65,6 +65,10 @@ public void testGetValueAsText() throws IOException {
             var text = parser.getValueAsText();
             assertThat(text, Matchers.notNullValue());
             assertTextRef(text.bytes(), "bar\"baz\"");
+            // Retrieve the value for a second time to ensure the last value is available
+            text = parser.getValueAsText();
+            assertThat(text, Matchers.notNullValue());
+            assertTextRef(text.bytes(), "bar\"baz\"");
```

Comment:
You might consider updating `testGetValueRandomized()` to also double-retrieve the value from the parser.